torchvision/transforms/functional.py

```diff
@@ -82,6 +82,54 @@ def to_tensor(pic):
         return img
 
 
+def as_tensor(pic):
+    """Convert a ``PIL Image`` or ``numpy.ndarray`` to tensor of same type.
+
+    See ``AsTensor`` for more details.
+
+    Args:
+        pic (PIL Image or numpy.ndarray): Image to be converted to tensor.
+
+    Returns:
+        Tensor: Converted image.
+    """
+    if not(_is_pil_image(pic) or _is_numpy(pic)):
+        raise TypeError('pic should be PIL Image or ndarray. Got {}'.format(type(pic)))
+
+    if _is_numpy(pic) and not _is_numpy_image(pic):
+        raise ValueError('pic should be 2/3 dimensional. Got {} dimensions.'.format(pic.ndim))
+
+    if isinstance(pic, np.ndarray):
+        # handle numpy array
+        if pic.ndim == 2:
+            pic = pic[:, :, None]
+
+        img = torch.from_numpy(pic.transpose((2, 0, 1)))
+        return img
+
+    if accimage is not None and isinstance(pic, accimage.Image):
```

Comment:
If we check in 96 if the image either `_is_pil_image` or `_is_numpy` will this ever be `True`?

Reply:
So if you refer to _is_pil_image on lines #15-17

`def _is_pil_image(img):
    if accimage is not None:
        return isinstance(img, (Image.Image, accimage.Image))`

It will pass the first two checks and then get to this check as is done in ToTensor()

Reply:
Dang. We really should change that name. I just assumed that it would check for `PIL` and be done for it.